services/src/api/model/datatypes.rs

```diff
@@ -1032,3 +1296,267 @@ impl TryFrom<SerializablePalette> for Palette {
+    fn from(_: geoengine_datatypes::primitives::NoGeometry) -> Self {
+        Self {}
+    }
+}
+
+impl From<NoGeometry> for geoengine_datatypes::primitives::NoGeometry {
+    fn from(_: NoGeometry) -> Self {
+        Self {}
+    }
+}
+
+impl ToSchema for NoGeometry {
+    fn schema() -> utoipa::openapi::Schema {
+        use utoipa::openapi::*;
+        ObjectBuilder::new()
+            .default(Some(serde_json::Value::Null))
+            .example(Some(serde_json::Value::Null))
+            .into()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct MultiPoint {
+    coordinates: Vec<Coordinate2D>,
+}
+
+impl From<geoengine_datatypes::primitives::MultiPoint> for MultiPoint {
+    fn from(value: geoengine_datatypes::primitives::MultiPoint) -> Self {
+        Self {
+            coordinates: value.points().iter().map(|x| (*x).into()).collect(),
+        }
+    }
+}
+
+impl From<MultiPoint> for geoengine_datatypes::primitives::MultiPoint {
+    fn from(value: MultiPoint) -> Self {
+        Self::new(value.coordinates.iter().map(|x| (*x).into()).collect()).unwrap()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct MultiLineString {
+    coordinates: Vec<Vec<Coordinate2D>>,
+}
+
+impl From<geoengine_datatypes::primitives::MultiLineString> for MultiLineString {
+    fn from(value: geoengine_datatypes::primitives::MultiLineString) -> Self {
+        Self {
+            coordinates: value
+                .lines()
+                .iter()
+                .map(|x| x.iter().map(|x| (*x).into()).collect())
+                .collect(),
+        }
+    }
+}
+
+impl From<MultiLineString> for geoengine_datatypes::primitives::MultiLineString {
+    fn from(value: MultiLineString) -> Self {
+        Self::new(
+            value
+                .coordinates
+                .iter()
+                .map(|x| x.iter().map(|x| (*x).into()).collect())
+                .collect(),
+        )
+        .unwrap()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct MultiPolygon {
+    polygons: Vec<Vec<Vec<Coordinate2D>>>,
+}
+
+impl From<geoengine_datatypes::primitives::MultiPolygon> for MultiPolygon {
+    fn from(value: geoengine_datatypes::primitives::MultiPolygon) -> Self {
+        Self {
+            polygons: value
+                .polygons()
+                .iter()
+                .map(|x| {
+                    x.iter()
+                        .map(|y| y.iter().map(|y| (*y).into()).collect())
+                        .collect()
+                })
+                .collect(),
+        }
+    }
+}
+
+impl From<MultiPolygon> for geoengine_datatypes::primitives::MultiPolygon {
+    fn from(value: MultiPolygon) -> Self {
+        Self::new(
+            value
+                .polygons
+                .iter()
+                .map(|x| {
+                    x.iter()
```

Comment:
use `into_iter` instead of `iter`

Reply:
[44637dd](https://github.com/geo-engine/geoengine/pull/662/commits/44637ddbac75b8e5ddfd9ab6d09f7776eb6cbdab)

Reply:
Calling `.into_iter()` here throws clippy errors like in the following so I reverted it. [11d3913](https://github.com/geo-engine/geoengine/pull/662/commits/11d3913d6d82700baa28c1e1338f6e9a0143489f)

```
error: this `.into_iter()` call is equivalent to `.iter()` and will not consume the `Vec`
    --> services/src/api/model/datatypes.rs:1464:36
     |
1464 |                         .map(|y| y.into_iter().map(|y| (*y).into()).collect())
     |                                    ^^^^^^^^^ help: call directly: `iter`
```